OpenQA/Isotovideo/Utils.pm

```diff
@@ -101,10 +104,18 @@ sub checkout_git_repo_and_branch ($dir_variable, %args) {
     my $clone_url = $url->fragment(undef)->to_string;
     my $local_path = $url->path->parts->[-1] =~ s/\.git$//r;
     my $tries = $args{retry_count};
-    while ($tries--) {
-        last unless clone_git $local_path, $clone_url, $args{clone_depth}, $branch, $dir, $dir_variable;
-        bmwqemu::diag "Clone failed, retries left: $tries of $args{retry_count}";
-        sleep($args{retry_interval} // 5);
+  retry: {
+        my $error;
+        do {
+            my $status;
+            eval { $status = clone_git $local_path, $clone_url, $args{clone_depth}, $branch, $dir, $dir_variable };
+            $error = $@;
+            last if $status;
+            bmwqemu::diag "Clone failed, retries left: $tries of $args{retry_count}";
+            rmdir $local_path;
+            sleep($args{retry_interval} // 5);
+        } while ($tries-- > 0);
+        die $error;
```

Comment:
This `die` is outside of the loop. Wouldn't we always run into this case?

Reply:
No, we hit it only if the `last` is not used in the inner loop. The `last` will end the named block, not the do-while.

Reply:
Ok, this makes sense. I'm still wondering whether we can simplify this. Having a label that is not really referenced and having two nested loops seems a bit too complicated for this use-case.

Reply:
I really tried, this construction was suggested on https://www.perltutorial.org/perl-do-while/

Reply:
I can probably use `last retry if $status;` if you want the label to be mentioned somewhere.

Reply:
Would it work to put this code into a sub-function and return from there instead of a loop with label?

Reply:
I managed to rearrange the code slightly so the second loop is no longer needed. It behave slightly differently (setting `$bmwqemu::vars{$dir_variable}` before it dies)  , hopefully it does not matter.